### DISCUSSION TITLE
I want to use useRouter under the app folder but report an error.

### DISCUSSION THREAD

**Discussion by chovrio** [vercel/next.js#45948] [Help] [answered]
Because I want to get the query parameter of the current page, I have to use useRouter, but the page file is under the app folder, referencing "next/router" will report an error, referencing "next/navigator" only has the method of operating the route.

      I have tried two methods. First, mark 'use client', then use location. It is possible in the development environment, but an error will be reported when packaging, because there is no window in the node environment.

      The second method is to write the pages that need to use useRouter under the pages folder. This method is feasible.

      But if I definitely want to get the query parameter or page url of the current page in the app folder, is there any way? I found no answer in the issues of next.js.

**Comment by icyJoseph** [comment, marked as answer]
`next/router` doesn't work under appDir, and `useRouter` from `'next/navigation` doesn't give you the query parameters anymore.

You have to use https://beta.nextjs.org/docs/api-reference/use-search-params, to get them as search parameters. 

And that has to be as a client component, from the docs:

```jsx
'use client';

import { useSearchParams } from 'next/navigation';

export default function Page() {
  const searchParams = useSearchParams();

  // E.g. `/dashboard?page=2&order=asc`
  const page = searchParams.get('page');
  const order = searchParams.get('order');

  return (
    <div>
      <p>Page: {page}</p>
      <p>Order: {order}</p>
    </div>
  );
}
```

**Reply by chovrio** [reply]
Thank you very much.

**Reply by icyJoseph** [reply]
If it works, don't forget to mark as answered :)
